### PR TITLE
Add a note about setting the value of maxNumConcurrentSubTasks 

### PR DESCRIPTION
**Pull request by ektravel**
### Description

This PR clarifies how to use the Combining input source with Druid delegate input sources.

### Issue

Attempting to use combining input source with two Druid delegate data sources results in failed ingestion.

### Solution

For combining input source to work correctly, the value of `maxNumConcurrentSubTasks` must be set to one of the following:
- more or equal to 1 for range or single_dim (Range and single_dim always use parallel mode)
 - more or equal to 2 for hashed or dynamic

This PR has:
- [X] been self-reviewed.

### CODE REVIEW

**Comment by FrankChen021** [comment]
Hi @ektravel , Could you paste the key information of the original issue in this PR? I think people out of Imply are not able to see that issue by your link.

**Comment by ektravel** [comment]
Hi @FrankChen021! I've updated the PR description with key information per your request.

File: docs/ingestion/native-batch-input-source.md [hunk 1]
```diff
@@ -707,18 +707,21 @@ Compared to the other native batch InputSources, SQL InputSource behaves differe
...
+Similar to other input sources, the Combining input source supports a single `inputFormat`.
+Delegate input sources that require an `inputFormat` must have the same format for input data.
+
+|Property|Description|Required|
 |--------|-----------|---------|
 |type|This should be "combining".|Yes|
-|delegates|List of _splittable_ InputSources to read data from.|Yes|
-
-Sample spec:
+|delegates|List of _splittable_ input sources to read data from.|Yes|
```
Comment (vtlim) [inline docs/ingestion/native-batch-input-source.md:722]:
```suggestion
|delegates|List of splittable input sources to read data from.|Yes|
```

File: docs/ingestion/native-batch-input-source.md [hunk 2]
```diff
@@ -748,3 +751,8 @@ Sample spec:
 ...
 ```
 
+For the Combining input source to read data correctly, set the value of `maxNumConcurrentSubTasks` in `tuningConfig` as follows:
```
Comment (vtlim) [inline docs/ingestion/native-batch-input-source.md:754]: Adds a bit more context
```suggestion
The [secondary partitioning method](native-batch.md#partitionsspec) determines the requisite number of concurrent worker tasks that run in parallel to complete ingestion with the Combining input source.
Set this value in `maxNumConcurrentSubTasks` in `tuningConfig` based on the secondary partitioning method:
```

File: docs/ingestion/native-batch-input-source.md [hunk 3]
```diff
@@ -707,18 +707,21 @@ Compared to the other native batch InputSources, SQL InputSource behaves differe
...
+It identifies the splits from delegate input sources and uses a worker task to process each split.
+Use the Combining input source only if all the delegates are _splittable_ and can be used by the [Parallel task](./native-batch.md). 
 
-|property|description|required?|
+Similar to other input sources, the Combining input source supports a single `inputFormat`.
+Delegate input sources that require an `inputFormat` must have the same format for input data.
+
+|Property|Description|Required|
 |--------|-----------|---------|
 |type|This should be "combining".|Yes|
```
[the remark this answers is not in the document]
Comment (ektravel) [reply docs/ingestion/native-batch-input-source.md:721]: Excellent suggestion!

File: docs/ingestion/native-batch-input-source.md [hunk 4]
```diff
@@ -738,7 +737,7 @@ Each of the SQL queries will be run in its own sub-task and thus for the above e
 
 **Recommended practices**
 
-Compared to the other native batch InputSources, SQL InputSource behaves differently in terms of reading the input data and so it would be helpful to consider the following points before using this InputSource in a production environment:
+Compared to the other native batch input sources, SQL input source behaves differently in terms of reading the input data and so it would be helpful to consider the following points before using this input source in a production environment:
```
Comment (techdocsmith) [inline docs/ingestion/native-batch-input-source.md:740]:
```suggestion
Compared to the other native batch input sources, SQL input source behaves differently in terms of reading the input data. Therefore, consider the following points before using this input source in a production environment:
```

**Review by techdocsmith — APPROVED** [review]
minor nit for clarity. Otherwise LGTM.

**Pull request merged**